goodtablesio/signals.py

```diff
@@ -17,3 +18,13 @@ def init_worker(**kwargs):
 def shutdown_worker(**kwargs):
     database['session'].bind.dispose()
     database['session'].close()
+
+
+@signals.task_failure.connect
+def task_failure(**kwargs):
+    exception = kwargs['exception']
+    if isinstance(exception, sqlalchemy.exc.SQLAlchemyError):
+        # To prevent session from break because of unhandled error with no rollback
+        # https://github.com/frictionlessdata/goodtables.io/issues/97
+        log.info('Database session rollback by celery error handler')
+        database['session'].rollback()
```

Comment:
About disposing session - do you think we need it? This process will be ended I suppose anyway after this code.